src/sonic-config-engine/sonic-cfggen

```diff
@@ -47,7 +48,9 @@ def unique_name(l):
 
 def main():
     parser=argparse.ArgumentParser(description="Render configuration file from minigraph data and jinja2 template.")
-    parser.add_argument("-m", "--minigraph", help="minigraph xml file")
+    group = parser.add_mutually_exclusive_group()
+    group.add_argument("-m", "--minigraph", help="minigraph xml file")
+    group.add_argument("-M", "--device-description", help="device description xml file")
```

Comment:
Don't see your point. Those are different files with different schema. I already make the output format consistence so you can use same tool and same template to generate, for example, `interfaces` file.